lib/go-atscfg/remapdotconfig.go

```diff
@@ -1161,27 +1161,27 @@ func makeFQDN(hostRegex string, ds *DeliveryService, server string, cdnDomain st
 		re = strings.Replace(re, `.*`, ``, -1)
 
 		hName := server
-		if ds.Type.IsDNS() {
-			if ds.RoutingName == nil {
+		if tc.DSType(*ds.Type).IsDNS() {
+			if &ds.RoutingName == nil || ds.RoutingName == "" {
```

Comment:
The `&ds.RoutingName == nil` check can be removed.

Reply:
removed and also added nil check
`if ds.Type != nil && tc.DSType(*ds.Type).IsDNS()`